x/rollapp/keeper/msg_server_update_state.go

```diff
@@ -34,17 +34,19 @@ func (k msgServer) UpdateState(goCtx context.Context, msg *types.MsgUpdateState)
 		return nil, errorsmod.Wrap(err, "before update state")
 	}
 
-	// verify the DRS version is not vulnerable
-	if k.IsDRSVersionVulnerable(ctx, msg.DrsVersion) {
-		// the rollapp is not marked as vulnerable yet, mark it now
-		err := k.MarkRollappAsVulnerable(ctx, msg.RollappId)
-		if err != nil {
-			return nil, fmt.Errorf("mark rollapp vulnerable: %w", err)
+	for _, bd := range msg.BDs.BD {
+		// verify the DRS version is not vulnerable
+		if k.IsDRSVersionVulnerable(ctx, bd.DrsVersion) {
+			// the rollapp is not marked as vulnerable yet, mark it now
+			err := k.MarkRollappAsVulnerable(ctx, msg.RollappId)
```

Comment:
ah I guess this is a store read
could do a happy path optimization to avoid rechecking ones which are already seenf
but IDK if we want to microoptimize